.github/workflows/version_bump.yml

```diff
@@ -0,0 +1,31 @@
+name: Version Bump
+
+on:
+  create: # It will trigger whenever a new tag or branch is created
+
+jobs:
+  version-bump:
+    if: startsWith(github.ref, 'refs/tags/') # true if new tag is created
+    runs-on: ubuntu-22.04
+    timeout-minutes: 3
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Bump version
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+          pnpm version ${GITHUB_REF#refs/tags/v} --no-git-tag-version 
+          git commit -am "Bump version"
+          git push origin main
```

Comment:
Since `main` is a protected branch, we would need to instead create a PR before merging into `main`.
We could use the https://github.com/marketplace/actions/create-pull-request action here to do the creating/updating of a PR. 

The `pnpm version ${GITHUB_REF#refs/tags/v} --no-git-tag-version` looks good though :+1: